contracts/settling_game/modules/combat/library.cairo

```diff
@@ -532,6 +442,121 @@ namespace Combat {
         return (updated_attacking_army,);
     }
 
+    // @notice calculates the health percentage loss of a battle. 450 is half the size of a fully maxxed Army
+    // @param outcome: battle outcome
+    // @return health_percentage: percentage loss
+    func calculate_health_loss_percentage{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr
+    }(outcome: felt) -> (health_percentage: felt) {
+        alloc_locals;
+
+        let less_than = is_le(outcome, 0);
+
+        if (less_than == TRUE) {
+            let (i, _) = unsigned_div_rem(((-outcome)) * 1000, 450);
+            return (health_percentage=1000 - i);
+        }
+
+        return (health_percentage=1000);
```

Comment:
Percentage of 1000 is weird 🤔 Does that mean 10x of something?

Reply:
it is for accuracy in the division, otherwise rounding errors are too brutal